File: my-folder/0349-intersection-of-two-arrays/solution.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        table = defaultdict(int)
        for num in nums1:
            table[num]+=1
        output = []
        keys = table.keys()
        for num in nums2:
            if num in keys and num not in output:
                output.append(num)
        return output

File: my-folder/0349-intersection-of-two-arrays/solution.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        set<int> s;
        for (auto n: nums1){
            s.insert(n);
        }
        vector<int> ans;

        for (auto num: nums2){
            if (s.find(num) != s.end())
                ans.push_back(num);
                s.erase(num);
        }
        return ans;

    }
};
